src/kafka-util/src/admin.rs

```diff
@@ -50,10 +63,10 @@ where
     if res.len() != 1 {
         return Err(CreateTopicError::TopicCountMismatch(res.len()));
     }
-    match res.into_element() {
-        Ok(_) | Err((_, RDKafkaErrorCode::TopicAlreadyExists)) => Ok(()),
-        Err((_, e)) => Err(CreateTopicError::Kafka(KafkaError::AdminOp(e))),
-    }?;
+    res.into_element().map(|_| ()).or_else(|e| match e {
+        (_, RDKafkaErrorCode::TopicAlreadyExists) if allow_existing => Ok(()),
+        (_, e) => Err(CreateTopicError::Kafka(KafkaError::AdminOp(e))),
+    })?;
```

Comment:
Doesn't much matter, but when you have to handle both the `Ok` and `Err` variants I usually find it a bit quicker to parse a `match` statement than stacked combinators:

```rust
    match res.into_element() {
        Ok(_) => (),
        Err((_, RDKafkaErrorCode::TopicAlreadyExists)) if allow_existing => (),
        Err((_, e)) => return Err(CreateTopicError::Kafka(KafkaError::AdminOp(e))),
    }
```